app/controllers/application_controller.rb

```diff
@@ -5,6 +5,130 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if session.has_key?(:id)
+      @user = User.find(session[:id])
+      redirect "/tweets"
+    else
+      erb :signup
+    end
+  end
+
+  get '/tweets' do
+    if session.has_key?(:id)
```

Comment:
You can use your `logged_in?` helper method here, which is preferable to referring to the session object directly.